src/backend/distributed/executor/multi_executor.c

```diff
@@ -45,12 +45,114 @@
  * TRUNCATE and real-time SELECT queries.
  */
 int MultiShardConnectionType = PARALLEL_CONNECTION;
+bool WritableStandbyCoordinator = false;
 
 
 /* local function forward declarations */
+static bool IsCitusPlan(Plan *plan);
+static bool IsCitusCustomScan(Plan *plan);
 static Relation StubRelation(TupleDesc tupleDescriptor);
 
 
+/*
+ * CitusExecutorStart is the ExecutorStart_hook that gets called when
+ * Postgres prepares for execution or EXPLAIN.
+ */
+void
+CitusExecutorStart(QueryDesc *queryDesc, int eflags)
+{
+	PlannedStmt *plannedStmt = queryDesc->plannedstmt;
+
+	if (RecoveryInProgress() && WritableStandbyCoordinator &&
+		IsCitusPlan(plannedStmt->planTree))
+	{
+		PG_TRY();
+		{
+			/* pretend that we're not in read-only mode to allow DML through */
+			XactReadOnly = false;
```

Comment:
d'oh, this is of those variables that we can't change on Windows.

Reply:
That'll be hard to fix without re-implementing `standard_ExecutorStart`. For now I'm disabling windows support.